contracts/src/TeleporterTokenSource.sol

```diff
@@ -166,11 +166,11 @@ abstract contract TeleporterTokenSource is
      * be already scaled to the local denomination for this token source.
      * Requirements:
      *
-     * - `input.destinationBlockchainID` cannot be the same as the current blockchainID
-     * - `input.destinationBridgeAddress` cannot be the zero address
-     * - `input.recipient` cannot be the zero address
-     * - `amount` must be greater than 0
-     * - `amount` must be greater than `input.primaryFee`
+     * - {input.destinationBlockchainID} cannot be the same as the current blockchainID
+     * - {input.destinationBridgeAddress} cannot be the zero address
+     * - {input.recipient} cannot be the zero address
+     * - {amount} must be greater than 0
+     * - {amount} must be greater than {input.primaryFee}
      */
     function _send(
         SendTokensInput memory input,
```

Comment:
can't comment on line, but if is multihop, we should check that fallback recipient is not zero address

Reply:
and if it isn't a multihop, might want to require fallback **is** zero address

Reply:
I don't think we need to check for a non-zero fallback recipient in the multi-hop case here actually because it would result in the intermediate message reverting irrevocably anyways, so we may as well just perform the fallback transfer anyways.

I'll add the extra requirement to the single hop case now.

Reply:
I don't think it necessarily is reverting irrevocably for the case the final destination is not registered/collateralized right? The intermediate step would revert, then after final destination is ready to send, a retry would get that through. But you're right that the `fallbackRecipient` at this point can't be changed anyways, so if a zero address somehow got through, though it's checked by default on `TeleporterTokenDestination`, nothing else we can do about it.

Can you add comments saying zero fallback recipient is checked on `TeleporterTokenDestination`, and since we can't change the value here etc. no need to check in addition.

Reply:
Yeah good point. As a corollary, we don't check that the `recipient` address is non-zero when receiving messages to mint/withdraw. I don't think we should include comments about `TeleporterTokenDestination` in the source contract either though because there is no guarantee that a specific destination implementation is used.